plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/util/DeltaLakeDomains.java

```diff
@@ -36,8 +36,12 @@ public static boolean partitionMatchesPredicate(Map<String, Optional<String>> pa
     {
         for (Map.Entry<DeltaLakeColumnHandle, Domain> enforcedDomainsEntry : domains.entrySet()) {
             DeltaLakeColumnHandle partitionColumn = enforcedDomainsEntry.getKey();
+            String column = partitionColumn.basePhysicalColumnName();
+            if (!partitionKeys.containsKey(column)) {
```

Comment:
Use `get` directly and null check on the result. Current code will look-up in the map twice when the entry is found